sound/soc/sof/loader.c

```diff
@@ -542,18 +541,21 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 	init_core_mask = ret;
 
 	/* now wait for the DSP to boot */
-	ret = wait_event_timeout(sdev->boot_wait, sdev->boot_complete,
+	ret = wait_event_timeout(sdev->boot_wait,
+				 sdev->fw_state != SOF_FW_STATE_BOOTING,
```

Comment:
this is odd, shouldn't this more fw_state == STATE_BOOT_OK?

Reply:
@plbossart this is the part that confused me too but we have a case where the FW booted but there's an ABI mismatch and I added a NOT_READY state for this. This is why I only check for != BOOTING

Reply:
you should make this explicit with a comment then. It looks like a bug when it's a feature :-)